### DISCUSSION TITLE
Specifying the Popstate event.

### DISCUSSION THREAD

**Discussion by batuhanbuena** [vercel/next.js#72901] [Help]
### Summary

Is there a way how i can identify between a forward or backwards action on a popstate event?. With the app router it overwrites the history state so i can only identify if there a popstate event but not if the user clicked the back or forward button.

**Comment by Mr-Meshky** [comment]
To detect the direction of movement (back or forward) in the popstate event, you can compare the previous and current state of the history. If `idx' is increasing in the history state, it means going forward and if it is decreasing, it means going backward.

```javascript
let lastState = window.history.state;

window.addEventListener("popstate", () => {
 const currentState = window.history.state;

 if (currentState.idx > lastState.idx) {
 console.log("front");
 } else if (currentState.idx < lastState.idx) {
 console.log("Back");
 }

 lastState = currentState;
});
```
I hope I have not misunderstood :)

**Reply by batuhanbuena** [reply]
This doesnt work because the state object is overwritten by the nextjs router and doesnt include the idx field.
